astro/terraform/terraform_remote_state_disable_utils.go

```diff
@@ -66,24 +69,51 @@ func deleteTerraformBackendConfig(in []byte) (updatedConfig []byte, err error) {
 		return nil, errors.New("could not parse \"terraform\" block in config")
 	}
 
-	if err := astDel(terraformConfigBlock.List, "backend"); err != nil {
-		return nil, err
-	}
+	astDelIfExists(terraformConfigBlock.List, "backend")
 
 	buf := &bytes.Buffer{}
 	printer.Fprint(buf, config)
 
 	return buf.Bytes(), nil
 }
 
-func deleteTerraformBackendConfigFromFile(file string) error {
+// hcl2 (used by terraform 0.12) doesn't provide interface to walk through the AST or
+// to modify block values, see https://github.com/hashicorp/hcl2/issues/23 and
+// https://github.com/hashicorp/hcl2/issues/88
+// As a work around we'll perform surgery directly on text, if backend config is simple.
+// The method returns an error, if the config is too complicated to be parsed with the regexp.
+// This method should be rewritten once hcl2 supports AST traversal and modification.
+func deleteTerraformBackendConfigWithHCL2(in []byte) (updatedConfig []byte, err error) {
+	// Regexp to find if any backend configuration exists
+	backendDefinitionRe := regexp.MustCompile(`(?s)[{\s+]backend\s+"[^"]+"\s*{`)
+	// Regexp to find simple backend configuration, which doesn't contain '{}' inside
+	backendBlockRe := regexp.MustCompile(`(?s)(\s*backend\s+"[^"]+"\s*{[^{]*?})`)
```

Comment:
It might be worth expanding this regexp and commenting it, e.g.:
https://stackoverflow.com/questions/24588352/go-idiom-for-writing-long-regular-expressions-embedded-comments

Reply:
Ack, thanks for the suggestion, I've added comments.